gce/burst-training/README.md

```diff
@@ -0,0 +1,242 @@
+In this case, although the dataset itself is small, the value of burst training
+is in allowing us to perform a much more comprehensive hyperparameter search
+compared to what would be possible on even a powerful laptop.
+
+
+## Code
+
+There are three types of code in this directory:
+
+1. Code facilitating the reproduction of your development environment on a
+   Compute Engine virtual machine.
+
+2. Code which defines your training job -- this is the kind of code that you
+   write every day using frameworks like [pandas](https://pandas.pydata.org/),
+   [scikit-learn](http://scikit-learn.org/stable/), and
+   [xgboost](https://github.com/dmlc/xgboost). In the case of this guide, this
+   is the code which directly pertains to the Census income problem.
+
+3. Code which executes your training job in a Compute Engine instance running
+   your development environment as specified in step 1.
+
+
+### Reproducing your environment on Compute Engine
+
+The very first step in the burst training process is making sure that Compute
+Engine is set up to execute the code that you either already have or will be
+writing locally. In the case of our Census example, we will be using Python 2.7
+with the packages listed in [requirements.txt](./requirements.txt).
+
+We do this by creating a virtual machine image that we can apply to any Compute
+Engine instances we create. An image is, essentially, the frozen disk state of a
+Compute Engine instance which, when used to create *another* instance, brings
+that instance up with the same state. This means that, if your image already had
+*scikit-learn* installed, then every virtual machine created with that image
+would come with *scikit-learn* out of the box.
+
+The burst training framework provides you with a script that you can run to
+build your image - [build-image.sh](./build-image.sh). This script brings up a
+VM instance, installs the appropriate libraries and packages, creates the image,
+and then deletes the Compute Engine instance when it is done.
+
+Packages are installed on the VM is done by means of a [startup
+script](https://cloud.google.com/deployment-manager/docs/step-by-step-guide/setting-metadata-and-startup-scripts),
+which is executed on the VM immediately after it starts. For the Census example,
+we are using [image.sh](./image.sh). In your own work, it should suffice to make
+the appropriate modifications to that file.
+
+You can run [build-image.sh](./build-image.sh) from your terminal:
+```
+./build-image.sh <INSTANCE-NAME> <IMAGE-FAMILY>
+```
+
+Here `<INSTANCE-NAME>` is a name you would like to assign to the temporary GCE
+instance on which we build the environment as specified in
+[image.sh](./image.sh). This instance will be deleted once the image has built,
+so you won't have to deal with it for very long.
+
+`<IMAGE-FAMILY>` starts a family of images under which Compute Engine stores
```

Comment:
suggestion: rephrase this from "image family" to something like "image versioning".

Reply:
"Image family" is the correct Compute Engine terminology, so I would prefer to keep it this way.

I feel it is better than adding another layer of terminology on top of the Compute Engine language.

Reply:
Sound reasonable to you @dizcology ?

Reply:
I agree that if that's the name GCE decide to use then we should probably keep it.